server/graphql/cache.js

```diff
@@ -0,0 +1,37 @@
+export const purgeCacheForCollectiveOperationNames = [
+  'CollectivePage',
+  'BudgetSection',
+  'TransactionsPage',
+  'ExpensesPage',
+  'CollectiveCover',
+  'CollectiveBannerIframe',
+];
+
+export function getGraphqlCacheKey(req) {
+  if (req.remoteUser) {
+    return;
+  }
+  if (!req.body || !req.body.operationName) {
+    return;
+  }
+  if (!req.body.variables || !req.body.variables.slug) {
+    return;
+  }
+  switch (req.body.operationName) {
+    case 'BudgetSection':
+    case 'CollectivePage':
+    case 'CollectiveCover':
+    case 'CollectiveBannerIframe':
+      return `${req.body.operationName}_${req.body.variables.slug}`;
+    case 'TransactionsPage':
+      if (req.body.variables.offset === 0 && req.body.variables.limit === 15) {
+        return `${req.body.operationName}_${req.body.variables.slug}`;
+      }
+      break;
+    case 'ExpensesPage':
+      if (req.body.variables.offset === 0 && req.body.variables.limit === 10) {
+        return `${req.body.operationName}_${req.body.variables.slug}`;
+      }
+      break;
```

Comment:
Many variables can influence the results on these pages (payout method type, status, amount, etc). How are we dealing with that if there's only one cache key?

Reply:
True, it's not properly implemented, the idea was to only cache the default case where you have slug+offset=0+limit=10.

I'm gonna change that.